test/groovy/util/JenkinsLoggingRule.groovy

```diff
@@ -12,19 +11,20 @@ import static org.junit.Assert.assertThat
 
 import org.hamcrest.Matchers
 
+
 class JenkinsLoggingRule implements TestRule {
 
     final BasePipelineTest testInstance
 
-    def expected = []
+    List<String> expected = []
```

Comment:
What was the issue with the def? Wouldn't a `Set` more appropriate since a `List` has duplicates and a `Set` not. If this is not used for succeeding that PR I would vote for keep it as is or make another PR for this.

Reply:
That was creating code style warnings and failed to IDE to match the types correctly. List<String> is just the explicit representation of def expected = [] which implicitly creates a List and in our scenario we add Strings to it.

Reply:
`Set expected = []` ends up in having a Set, which is more appropriate since Sets does not have duplicates, Lists have. I agree, this is nit-pick, but if we improve code (type-safe rather than generic `def`) we should do it right. Having refactoring only PRs rather than doing it a s side-thread in another PR might help getting things merged faster.